From 2a2fbce97aaa10a0e2f53f1cf7014cad67e808a0 Mon Sep 17 00:00:00 2001
From: Thomas Prescher <thomas.prescher@cyberus-technology.de>
Date: Tue, 6 Feb 2024 11:25:35 +0100
Subject: [PATCH 07/17] NEM: KVM: disable KVM access to Hyper-V MSRs

If we want KVM to handle these MSRs, we also need to sync them correcly in
nemHCLnxImportState and nemHCLnxExportState. Otherwise, VM reboot and
state save/resume will be broken for guests that utilize the HyperV PV interface
because the MSRs go out of sync.
---
 src/VBox/VMM/VMMR3/NEMR3Native-linux.cpp | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/src/VBox/VMM/VMMR3/NEMR3Native-linux.cpp b/src/VBox/VMM/VMMR3/NEMR3Native-linux.cpp
index 38251a5568..1077864943 100644
--- a/src/VBox/VMM/VMMR3/NEMR3Native-linux.cpp
+++ b/src/VBox/VMM/VMMR3/NEMR3Native-linux.cpp
@@ -1112,6 +1112,13 @@ int nemR3NativeInitCompleted(PVM pVM, VMINITCOMPLETED enmWhat)
         /** @todo add more? */
         MSR_RANGE_END(64);
 
+/*
+ * FIXME: If we want KVM to handle these MSRs, we also need to sync them correctly in
+ * nemHCLnxImportState and nemHCLnxExportState. Otherwise, VM reboot and
+ * state save/resume will be broken for guests that utilize the HyperV PV interface
+ * because the MSRs go out of sync.
+ */
+#if 0
         if (pVM->gim.s.enmProviderId == GIMPROVIDERID_HYPERV)
         {
             MSR_RANGE_BEGIN(0x40000000, 0x40003000, KVM_MSR_FILTER_READ | KVM_MSR_FILTER_WRITE);
@@ -1154,6 +1161,7 @@ int nemR3NativeInitCompleted(PVM pVM, VMINITCOMPLETED enmWhat)
 
             MSR_RANGE_END(64);
         }
+#endif
 
         /** @todo Specify other ranges too? Like hyper-V and KVM to make sure we get
          *        the MSR requests instead of KVM. */
-- 
2.43.0

